cocos/animation/marionette/state-machine/state-machine-eval.ts

```diff
@@ -1498,15 +1502,16 @@ class ConditionEvaluationContextImpl implements ConditionEvaluationContext {
     public get sourceStateMotionTimeNormalized () {
         const { _sourceState: sourceState } = this;
         assertIsTrue(
-            sourceState && isRealState(sourceState) && sourceState.activeReferenceCount,
-            `State motion time is only defined on activated motion states, pose states and empty states.`,
+            sourceState
+            && (sourceState.kind === NodeKind.animation || sourceState.kind === NodeKind.procedural)
+            && sourceState.activeReferenceCount,
+            `State motion time is only defined on activated motion states and procedural pose states.`,
         );
         switch (sourceState.kind) {
         case NodeKind.animation:
-            return sourceState.time;
+            return sourceState.normalizedTime;
         case NodeKind.procedural:
-            // TODO:
-            // fallthrough
+            return sourceState.countMotionTime();
```

Comment:
How does `countMotionTime` calculate time?

Reply:
I mentioned in PR content:

> - If the pose graph has no any motion player nodes(ie. PoseNodePlayMotion nodes), the state's SMT is always zero.

> - Otherwise, if the pose graph has exactly one motion player node, the state's SMT is the motion's normalized play time.

> - Otherwise, the pose graph has more than one motion player nodes, the state's SMT is the normalized play time of motion player node **having max weight**.